src/MemoryTable.js

```diff
@@ -7,28 +8,67 @@ import ContextureMobx from './utils/contexture-mobx'
 import { componentForType } from './utils/schema'
 import { ResultTable, TypeMap } from './exampleTypes'
 
-export let memoryService = (records, { schema, debug } = {}) =>
-  Contexture({
-    debug,
-    // Hack to effectively set a default schema: if our tree root doesn't have
-    // a `schema` property, it will get the schema at key `undefined`.
-    schemas: { [schema]: { memory: { records } } },
-    providers: { memory: { ...memory, types: types() } },
-  })
-
-let MemoryTable = ({ data, fields, debug, include, ...props }) => {
-  let service = memoryService(data, { schema: 'data', debug })
+export let useMemoryTree = ({
+  records,
+  schema = 'data',
+  fields,
+  debug,
+  resultNode = {
+    pageSize: 50,
+  },
+  criteriaNodes = [],
+  childrenNodes = [],
+} = {}) => {
+  let include = _.map('field', fields)
+  let [memoryStorage] = React.useState({ records: [] })
   let [tree] = React.useState(
-    ContextureMobx({ service })({
+    ContextureMobx({
+      disableAutoUpdate: true,
+      service: Contexture({
+        debug,
+        schemas: { [schema]: { memory: memoryStorage } },
+        providers: { memory: { ...memory, types: types() } },
+      }),
+    })({
       key: 'root',
       schema: 'data',
       children: [
-        { key: 'results', type: 'results', include },
-        { key: 'criteria', type: 'group', children: [] },
+        { key: 'results', type: 'results', include, ...resultNode },
+        { key: 'criteria', type: 'group', children: criteriaNodes },
+        ...childrenNodes,
       ],
     })
   )
-  tree.refresh(['root'])
+
+  if (records !== memoryStorage.records) {
+    let updateMemory = async records => {
+      memoryStorage.records = await records
+      tree.refresh(['root'])
+    }
+    updateMemory(records)
+  }
+
+  return tree
+}
+
+let MemoryTable = ({
```

Comment:
Can't comment on the code at the end of this file but I'm pretty sure this component doesn't need to be an observer.